### PR TITLE
Extension web pack

### PR DESCRIPTION
**Pull request by Tatsinnit**
This PR aims for the enable web pack. Very early eye balling Draft PR for web pack. 

~Currently there is one warning regarding: `applicationinsights-native-metrics` which I could resolve it using : `npm i --save applicationinsights-native-metrics` but then it end up with the unexpected character error as mentioned here:~

* https://github.com/microsoft/ApplicationInsights-node.js-native-metrics/issues/4
* (Fix for this got merged not long ago so I am not sure if this is released yet) :  https://github.com/microsoft/ApplicationInsights-node.js-native-metrics/pull/8

Should take care of these:  

* https://github.com/Azure/vscode-aks-tools/issues/17
* https://github.com/Azure/vscode-aks-tools/issues/54

### CODE REVIEW

File: src/commands/utils/reporter.ts [hunk 1]
```diff
@@ -19,8 +19,8 @@ interface IPackageInfo {
     aiKey: string;
 }
 
-function getPackageInfo(context: vscode.ExtensionContext): IPackageInfo | undefined {
-    const extensionPackage = require(context.asAbsolutePath('./package.json'));
+function getPackageInfo(_context: vscode.ExtensionContext): IPackageInfo | undefined {
+    const extensionPackage = require('../../../package.json'); // require(context.asAbsolutePath('./package.json'));
```
Comment (itowlson) [inline src/commands/utils/reporter.ts:23]: You can probably drop the comment at this point?

File: src/tree/aksClusterTreeItem.ts [hunk 2]
```diff
@@ -3,9 +3,10 @@ import { Resource } from "azure-arm-storage/lib/models";
 import { CloudExplorerV1 } from "vscode-kubernetes-tools-api";
 import { SubscriptionClient } from 'azure-arm-resource';
 import { toSubscription } from "../azure-api-utils";
-import { getExtensionPath } from "../commands/utils/host";
-import * as path from 'path';
-import * as vscode from 'vscode';
+// import { getExtensionPath } from "../commands/utils/host";
```
Comment (itowlson) [inline src/tree/aksClusterTreeItem.ts:6]: Drop commented-out code

File: src/tree/aksClusterTreeItem.ts [hunk 3]
```diff
@@ -23,7 +24,7 @@ export default class AksClusterTreeItem extends AzureTreeItem implements AksClus
         private readonly resource: Resource) {
         super(parent);
 
-        this.iconPath = vscode.Uri.file(path.join(getExtensionPath()!, 'resources', 'aks-tools.png'));
+        this.iconPath = assetUri('resources/aks-tools.png'); // vscode.Uri.file(path.join(getExtensionPath()!, 'resources', 'aks-tools.png'));
```
Comment (itowlson) [inline src/tree/aksClusterTreeItem.ts:27]: Commented out code

File: webpack.config.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+//@ts-check
+
+'use strict';
+
+const path = require('path');
+const webpack = require('webpack');
+
+/**@type {import('webpack').Configuration}*/
+const config = {
+  target: 'node', // vscode extensions run in a Node.js-context 📖 -> https://webpack.js.org/configuration/node/
```
Comment (itowlson) [inline webpack.config.js:10]: Funny icon in the comment...?

File: webpack.config.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  },
+  devtool: 'source-map',
+  externals: {
+    vscode: 'commonjs vscode',
+    bufferutil: 'commonjs bufferutil',
+    'spawn-sync': 'commonjs spawn-sync',
+    'utf-8-validate': 'commonjs utf-8-validate'
+  },
+  plugins: [
+      new webpack.IgnorePlugin(/^\.\/locale$/, /handlebars$/)
```
Comment (itowlson) [inline webpack.config.js:28]: I ended up with an ignore for `electron` too - not sure if that is useful / necessary or just vestigial from one of my many dead ends.

**Review by itowlson — COMMENTED** [review]
This looks good and my comments are only cosmetic - nice work!  One thing I would strongly recommend is creating a VSIX package (using `vsce`) and installing that and testing it thoroughly!  In particular keep an eye out for any missing assets (e.g. icons) and check that the telemetry is still working (because of the change to `getPackageInfo`).  Looking forward to this going out - thank you!

**Comment by Tatsinnit** [comment]
💡 Please note: Closing this and I will reopen as another PR because we have the Deprecation of SDK package work sorted as well. Thank you.

**Pull request closed without merging**